autoload/denops_shared_server.vim

```diff
@@ -10,35 +10,60 @@ function! denops_shared_server#install() abort
         \ 'hostname': hostname,
         \ 'port': port,
         \}
-  if executable('launchctl')
-    call denops_shared_server#launchctl#install(options)
-  elseif executable('systemctl')
-    call denops_shared_server#systemctl#install(options)
-  elseif has('win32')
-    call denops_shared_server#winsw#install(options)
-  else
-    call denops#util#error('This platform is not supported. Please configure denops-shared-server manually.')
+  try
+    if has('win32') && executable('powershell.exe')
+      call denops_shared_server#runtray#install(options)
+    elseif executable('launchctl')
+      call denops_shared_server#launchctl#install(options)
+    elseif executable('systemctl')
+      call denops_shared_server#systemctl#install(options)
+    else
+      throw printf('This platform is not supported. Please configure denops-shared-server manually.')
+    endif
+  catch /^\(Vim\)\@!/
+    call denops#util#error(v:exception)
     return
-  endif
-  call denops#util#info('wait 1 second for the shared server startup...')
-  sleep 1
+  endtry
+  call denops#util#info('wait 5 second for the shared server startup...')
+  sleep 5
   call denops#util#info('connect to the shared server')
   call denops#server#connect(g:denops_server_addr)
   call denops#util#info('stop the local server')
   call denops#server#stop()
 endfunction
 
 function! denops_shared_server#uninstall() abort
-  if executable('launchctl')
-    call denops_shared_server#launchctl#uninstall()
-  elseif executable('systemctl')
-    call denops_shared_server#systemctl#uninstall()
-  elseif has('win32')
-    call denops_shared_server#winsw#uninstall()
-  else
-    call denops#util#error('This platform is not supported. Please configure denops-shared-server manually.')
+  try
+    if has('win32') && executable('powershell.exe')
+      call denops_shared_server#runtray#uninstall()
+    elseif executable('launchctl')
+      call denops_shared_server#launchctl#uninstall()
+    elseif executable('systemctl')
+      call denops_shared_server#systemctl#uninstall()
+    else
+      throw printf('This platform is not supported. Please configure denops-shared-server manually.')
+    endif
+  catch /^\(Vim\)\@!/
+    call denops#util#error(v:exception)
     return
+  endtry
+endfunction
+
+function! denops_shared_server#_system(cmd, ...) abort
+  let l:opts = get(a:000, 0, {})
+  echomsg printf('>>> %s', a:cmd)
+  if has_key(l:opts, 'input')
+    let l:output = system(a:cmd, l:opts.input)
+  else
+    let l:output = system(a:cmd)
+  endif
+  for l:line in split(l:output, '\n')
+    echomsg l:line
+  endfor
+  if v:shell_error
+    throw printf('%s: code=%d', get(l:opts, 'error_message', 'Error'), v:shell_error)
   endif
+  return l:output
 endfunction
```

Comment:
This function is only used from `denops_shared_server/runtray.vim` thus it should be placed there (and maybe it should be a script local function)

Reply:
This function is named `_system` but the behavior is different from the builtin `system()` (for example, `system()` return string but this function also `echomsg` the result). It's confused. I feel that all `echomsg` things exist in the function are for debugging and if you remove these debugging code, the function become more simple.

Reply:
Removed `denops_shared_server#_system()` and use `system()`.